samza-core/src/main/java/org/apache/samza/clustermanager/container/placement/ContainerPlacementMessageObjectMapper.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.clustermanager.container.placement;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import org.apache.samza.container.placement.ContainerPlacementMessage;
+import org.apache.samza.container.placement.ContainerPlacementRequestMessage;
+import org.apache.samza.container.placement.ContainerPlacementResponseMessage;
+import org.codehaus.jackson.JsonGenerator;
+import org.codehaus.jackson.JsonNode;
+import org.codehaus.jackson.JsonParser;
+import org.codehaus.jackson.ObjectCodec;
+import org.codehaus.jackson.Version;
+import org.codehaus.jackson.map.DeserializationContext;
+import org.codehaus.jackson.map.JsonDeserializer;
+import org.codehaus.jackson.map.JsonSerializer;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.codehaus.jackson.map.SerializerProvider;
+import org.codehaus.jackson.map.jsontype.NamedType;
+import org.codehaus.jackson.map.module.SimpleModule;
+
+/**
+ * Object mapper for serializing and deserializing Container placement messages
+ */
+public class ContainerPlacementMessageObjectMapper {
+
+  private ContainerPlacementMessageObjectMapper() {
+  }
+
+  static ObjectMapper getObjectMapper() {
+    ObjectMapper objectMapper = new ObjectMapper();
+    SimpleModule module = new SimpleModule("ContainerPlacementModule", new Version(1, 0, 0, ""));
+    module.addSerializer(ContainerPlacementMessage.class,
+        new ContainerPlacementMessageObjectMapper.ContainerPlacementMessageSerializer());
+    module.addDeserializer(ContainerPlacementMessage.class,
+        new ContainerPlacementMessageObjectMapper.ContainerPlacementMessageDeserializer());
+    objectMapper.registerModule(module);
+    objectMapper.registerSubtypes(new NamedType(ContainerPlacementResponseMessage.class));
+    objectMapper.registerSubtypes(new NamedType(ContainerPlacementRequestMessage.class));
+    return objectMapper;
+  }
```

Comment:
Can we have a static ObjectMapper? It is thread safe and we are not changing its configuration either.

Reply:
I suppose you mean to have a static object Mapper declared here: https://github.com/apache/samza/pull/1227/files#diff-20fafbc969e1be36e38d63017cbfc77aR58?

Reply:
Yes and no, we can change that to be a static field or getObjectMapper() can return a static singleton as well.

Since getObjectMapper() takes no argument, so it is returning a object mapper with the same configuration, we could make getObjectMapper() to return the same instance of ObjectMapper instead of relying the caller to have a static variable.

Reply:
I see sure since we are not changing the configuration it can be a singleton